lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -165,7 +167,23 @@ try {
         Exit-Json $result
     }
 
-    $principal = New-ScheduledTaskPrincipal -UserId "$user" -LogonType ServiceAccount
+    # Handle RunAs/RunLevel options for the task
+
+    if (!$store_password) {
+        # Create a ScheduledTaskPrincipal for the task to run under
+        $principal = New-ScheduledTaskPrincipal -UserId $user -LogonType S4U -RunLevel $runlevel -Id Author
+        $registerRunOptionParams = @{Principal = $principal}
+    }
+    else {
+        # Specify direct credential and run-level values to add to Register-ScheduledTask
+        $registerRunOptionParams = @{
+            User = $user
+            RunLevel = $runlevel
+        }
+        if ($password) {
```

Comment:
It shouldn't, in this case - this would allow someone to set only `user` and `runlevel` and not explicitly set `store_password` to no, though it will effectively do just that.

221 definitely has some complicated logic to it - if you can think of a better way to refactor that portion and this one (172), let me know.